### DISCUSSION TITLE
How can i release snapshot for PR packages only?

### DISCUSSION THREAD

**Discussion by igoldny** [changesets/changesets#1083] [Q&A]
We have in our repo PR snapshot release workflow.
Each PR publish snapshot release for each commit (once all checks passed).

I would like to publish a snapshot release of only the changes within PR,
but if i have unpublished changesets on master PR snapshot release will include them as well.

What's the right way to publish snapshot release of the changes that i have in PR only without the changes
i have in `.changeset` folder taken from master branch?

**Comment by igoldny** [comment]
as a workaround solution i did the following:
```
sh "rm -rf .changeset/*.md"
sh "git diff --name-only origin/master..HEAD | grep .changeset/ | xargs git checkout --"
sh "yarn changeset version --snapshot ${snapshot}"
sh "git checkout .changeset"
```

i wonder if there is a better solution built in changeset?

**Comment by Andarist** [comment]
The problem with the approach that you want to achieve e is that often the packages touched in a PR depend on other packages that were touched previously. We absolutely have to publish those packages as well otherwise your published packages could easily end up being unusable. 

We could limit what is being published though if nothing touched in the PR relates to a given package anyhow. That still wouldn't be exactly the same as "PR packages only" though.

I'm open for accepting a PR that would implement this with some flag (`--since`?) or something.
